matrix_content_scanner/mcs.py

```diff
@@ -46,10 +46,11 @@ class MatrixContentScanner:
     def __init__(
         self,
         config: MatrixContentScannerConfig,
-        reactor: Reactor = twisted.internet.reactor,  # type: ignore[assignment]
     ) -> None:
         self.config = config
-        self.reactor = reactor
+        # Always use Twisted's asyncio reactor, because we need to use Futures in the
+        # scanner.
+        self.reactor = asyncioreactor.AsyncioSelectorReactor()
```

Comment:
you may want to 'install' the asyncio reactor, since some of Twisted's helpers use the globally-installed reactor (or at least, did when I last checked). I'm not sure what the repercussions are of mixing and matching reactors, but it doesn't sound like it can end well. 

That said, if you want to avoid it, I'm not fussed. I'm not a fan of the global state approach and maybe this means you can be a bit more careful about not accidentally relying on the global one.

Reply:
I think you're right. I went this way because I was getting `reactor already installed` errors, but after looking into it further it looks like it's because the content scanner was importing `twisted.internet.reactor` directly, which caused the default reactor to be already installed.